File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.js
/**
 * @param {number[]} chalk
 * @param {number} k
 * @return {number}
 */
var chalkReplacer = function (chalk, k) {
    let tot = 0
    chalk.forEach((x) => {
        tot += x
    })
    let re_after_rounds = k % tot
    for (let i = 0; i < chalk.length; ++i) {
        if (re_after_rounds < chalk[i])
            return i;
        re_after_rounds -= chalk[i]
    }
    return -1
};

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long can = 0;
        for (int x : chalk)
            can += x;
        k %= can;
        for (int i = 0; i < chalk.length; i++) {
            if (chalk[i] > k)
                return i;
            k -= chalk[i];
        }
        return -1; // No use of this actually
    }
}